From 756743e035b5244cbcea8b2117f11d8934aa1a98 Mon Sep 17 00:00:00 2001
From: Tatsuya Matsuo <96454784+matsuo-tat@users.noreply.github.com>
Date: Tue, 20 Feb 2024 13:31:55 +0900
Subject: [PATCH 08/11] [xcvrd] Fix the hostlanes to set datapath init fields

This change always sets DPDeinitLane and ApplyDPInitLane fields for all
hostlanes. This allows for consistent datapath state machine and
configuration between the hostlanes.
---
 sonic-xcvrd/xcvrd/xcvrd.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/sonic-xcvrd/xcvrd/xcvrd.py b/sonic-xcvrd/xcvrd/xcvrd.py
index 56efc3c..32e3cdc 100644
--- a/sonic-xcvrd/xcvrd/xcvrd.py
+++ b/sonic-xcvrd/xcvrd/xcvrd.py
@@ -873,6 +873,7 @@ class CmisManagerTask(threading.Thread):
     CMIS_DEF_EXPIRED     = 60 # seconds, default expiration time
     CMIS_MODULE_TYPES    = ['QSFP-DD', 'QSFP_DD', 'OSFP', 'QSFP+C']
     CMIS_MAX_HOST_LANES    = 8
+    CMIS_ALL_HOST_LANES_MASK = 0xff
 
     CMIS_STATE_UNKNOWN   = 'UNKNOWN'
     CMIS_STATE_INSERTED  = 'INSERTED'
@@ -1600,7 +1601,7 @@ class CmisManagerTask(threading.Thread):
                         self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_DEINIT
                     elif state == self.CMIS_STATE_DP_DEINIT:
                         # D.2.2 Software Deinitialization
-                        api.set_datapath_deinit(host_lanes_mask)
+                        api.set_datapath_deinit(self.CMIS_ALL_HOST_LANES_MASK)
 
                         # D.1.3 Software Configuration and Initialization
                         media_lanes_mask = self.port_dict[lport]['media_lanes_mask']
@@ -1665,7 +1666,7 @@ class CmisManagerTask(threading.Thread):
 
                         # D.1.3 Software Configuration and Initialization
                         api.set_application(host_lanes_mask, appl, ec)
-                        if not api.scs_apply_datapath_init(host_lanes_mask):
+                        if not api.scs_apply_datapath_init(self.CMIS_ALL_HOST_LANES_MASK):
                             self.log_notice("{}: unable to set application and stage DP init".format(lport))
                             self.force_cmis_reinit(lport, retries + 1)
                             continue
@@ -1696,7 +1697,7 @@ class CmisManagerTask(threading.Thread):
                             continue
 
                         # D.1.3 Software Configuration and Initialization
-                        api.set_datapath_init(host_lanes_mask)
+                        api.set_datapath_init(self.CMIS_ALL_HOST_LANES_MASK)
                         dpInitDuration = self.get_cmis_dp_init_duration_secs(api)
                         self.log_notice("{}: DpInit duration {} secs".format(lport, dpInitDuration))
                         self.port_dict[lport]['cmis_expired'] = now + datetime.timedelta(seconds=dpInitDuration)
-- 
2.20.1

